unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx

```diff
@@ -0,0 +1,131 @@
+
+interface ReceiptBaseFormProps {
+  lockAddress: string
+  network: number
+  isManager: boolean
+  disabled: boolean
+}
+export const ReceiptBaseForm = ({
+  lockAddress,
+  isManager,
+  disabled,
+  network,
+}: ReceiptBaseFormProps) => {
+  const {
+    register,
+    handleSubmit,
+    getValues,
+    reset,
+    formState: { isValid },
+  } = useForm<SupplierBodyProps>({
+    mode: 'onChange',
+  })
+
+  const getReceiptBase = async (): Promise<any> => {
+    return await storage.getReceiptsBase(network, lockAddress)
+  }
+
+  const saveReceiptBase = async (): Promise<any> => {
+    if (!isManager) return
+    const supplier = await getValues()
+    return await storage.saveReceiptsBase(network, lockAddress, {
+      data: {
+        ...supplier,
+      },
+    })
+  }
+
+  const receiptBaseMutation = useMutation(saveReceiptBase)
+
+  const onHandleSubmit = async () => {
+    if (isValid) {
+      await ToastHelper.promise(receiptBaseMutation.mutateAsync(), {
+        loading: 'Updating  supplier details,',
+        success: 'Supplier updated',
+        error: 'We could not update the details.',
+      })
+    } else {
+      ToastHelper.error('Form is not valid')
+      reset()
+    }
+  }
+
+  const { isLoading: isLoadingDetails } = useQuery(
+    ['getReceiptBase', lockAddress, network, receiptBaseMutation.isSuccess],
+    async () => {
+      return getReceiptBase()
+    },
+    {
+      onSuccess: (res) => {
+        if (res.data) {
+          reset(res.data)
+        }
+      },
+      enabled: isManager,
+    }
+  )
+
+  const disabledInput =
+    receiptBaseMutation.isLoading || disabled || isLoadingDetails
+
+  return (
+    <form
+      className="flex flex-col gap-6 text-left"
+      onSubmit={handleSubmit(onHandleSubmit)}
+    >
+      <div className="grid grid-cols-2 gap-4">
+        <Input
+          disabled={disabledInput}
+          label="Supplier name"
+          {...register('supplierName')}
+        />
+        <Input disabled={disabledInput} label="VAT" {...register('vat')} />
+
+        <div className="col-span-2">
+          <TextBox
+            disabled={disabledInput}
+            label="Service performed"
+            {...register('servicePerformed')}
+          />
+        </div>
+        <Input
+          disabled={disabledInput}
+          label="Address line 1"
+          {...register('addressLine1')}
+        />
+        <Input
+          disabled={disabledInput}
+          label="Address line 2"
+          {...register('addressLine2')}
+        />
+        <Input disabled={disabledInput} label="City" {...register('city')} />
+        <Input disabled={disabledInput} label="State" {...register('state')} />
+        <Input disabled={disabledInput} label="Zip" {...register('zip')} />
+        <Input
+          disabled={disabledInput}
+          label="Country"
+          {...register('country')}
+        />
+      </div>
+
```

Comment:
No error={errors.abc.message} passed in any? 

Should pass it even if we do not have any validation yet. We should add some here.

Reply:
All the fields are optional for now